geniza/corpus/management/commands/import_data.py

```diff
@@ -336,3 +372,153 @@ def log_creation(self, *objects):
+
+        # person with given name(s) and last name – case-insensitive lookup
+        if " " in name:
+            given_names, last_name = [sname.strip(punctuation) for sname in
+                                      name.rsplit(" ", 1)]
+            try:
+                user = User.objects.get(first_name__iexact=given_names,
+                                        last_name__iexact=last_name)
+            except User.DoesNotExist:
+                pass
+
+        # initials; use first & last to do lookup
+        elif name:
+            first_i, last_i = name[0], name[-1]
+            try:
+                user = User.objects.get(first_name__startswith=first_i,
+                                        last_name__startswith=last_i)
+            except User.DoesNotExist:
+                pass
+
+        # if we didn't get anyone through either method, warn and use team user
+        if not user:
+            logging.warning(
+                f"Couldn't find user {name} on PGPID {pgpid}; using {self.team_user}")
+            return self.team_user
+
+        # otherwise add to the cache using requested name and return the user
+        logging.debug(f"Found user {user} for {name} on PGPID {pgpid}")
+        self.user_lookup[name] = user
+        return user
+
+    def get_edit_history(self, input_by, date_entered, pgpid=None):
+        """Parse spreadsheet "input by" and "date entered" columns to
+        reconstruct the edit history of a single document.
+
+        Output is a list of dict to pass to log_edit_history. Each event has
+        a type (creation or revision), associated user, and timestamp.
+        """
+
+        # split both fields by semicolon delimiter & remove whitespace
+        all_input = map(str.strip, input_by.split(";"))
+        all_dates = map(str.strip, date_entered.split(";"))
+
+        # try to map every "input by" listing to a user account
+        users = []
+        for input_by in all_input:
+
+            # special case: two people together; add as a tuple
+            if " and " in input_by:
+                users.append(
+                    tuple(map(self.get_user, input_by.split(" and "))))
+            else:
+                users.append(self.get_user(input_by))
+
+        # convert every "date entered" listing to a date object. if any parts of
+        # the date are missing, fill with default values below. for details:
+        # https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.parse
+        dates = []
+        for date in all_dates:
+            try:
+                dates.append(parse(date, default=DEFAULT_EVENT_DATE).date())
+            except ParserError:
+                continue
+
+        # make sure we have same number of users/dates by padding with None
+        while len(users) < len(dates):
+            users.insert(0, None)
+
+        # moving backwards in time, pair dates with users and event types
+        events = []
+        for i in range(len(dates) - 1, -1, -1):
```

Comment:
This seems more complicated than it needs to be! Am having trouble following the logic.  Can't you just zip the two lists together? You can sort the dates independently to get the creation date if it isn't reliably first in the spreadsheet.